errata_parser.rb

```diff
@@ -267,29 +269,39 @@ def load_config(filename)
     fatal('Config-error missing \'repository\'-section', 4) unless cfg.key?('repository') && cfg['repository'].is_a?(Hash)
     fatal('Config-error \'repo_url\' missing in \'repository\'', 5) unless cfg['repository'].key? 'repo_url'
     fatal('Config-error \'releases\' missing in \'repository\'', 6) unless cfg['repository'].key? 'releases'
-    cfg['repository']['releases'].each do |s|
-      threads << Thread.new do
-        warn "START  Download #{s.inspect} from #{cfg['repository']['repo_url']}" if options[:verbose]
-        deb_rel = DebRelease.new(cfg['repository']['repo_url'], s)
-        deb_rel.release_name = fix_release(deb_rel.release_name, cfg['aliases']) if deb_rel.release_name.include? '-'
-        deb_rel.whitelist_comp = get_whitelist(cfg, 'components')
-        deb_rel.whitelist_arch = whitelist_arch
-        pkgs = deb_rel.all_packages
 
-        # merge package-list
-        mutex.synchronize do
-          pkgs.each_value do |pkg|
-            pkg.each do |arch_name, arch_pkgs|
-              arch_pkgs.each do |arch_pkg|
-                packages[arch_name] = {} unless packages.key? arch_name
-                packages[arch_name][arch_pkg['release']] = {} unless packages[arch_name].key? arch_pkg['release']
-                packages[arch_name][arch_pkg['release']][arch_pkg['name']] = [] unless packages[arch_name][arch_pkg['release']].key? arch_pkg['name']
-                packages[arch_name][arch_pkg['release']][arch_pkg['name']] << arch_pkg['version']
-              end
+    [cfg['repository'], cfg['repository-esm']].each do |repository|
+      next if repository.nil?
+
+      repository['releases'].each do |s|
+        threads << Thread.new do
+          begin
+            Thread.current[:repo_url] = repository['repo_url']
+            unless repository.fetch('repo_user', '').empty?
+              url = URI.parse(Thread.current[:repo_url])
+              url.user = repository['repo_user']
+              url.password = repository['repo_pass']
+              Thread.current[:repo_url] = url.to_s
+            end
+            warn "START  Download #{s.inspect} from #{Thread.current[:repo_url].sub(/:[^:@]+@/, ':*****@')}" if options[:verbose]
+            deb_rel = DebRelease.new(Thread.current[:repo_url], s)
+            deb_rel.release_name = fix_release(deb_rel.release_name, cfg['aliases']) if deb_rel.release_name.include? '-'
+            deb_rel.whitelist_comp = get_whitelist(cfg, 'components')
+            deb_rel.whitelist_arch = whitelist_arch
+            pkgs = deb_rel.all_packages
+
+            # merge package-list
+            mutex.synchronize do
+              DebRelease.assemble_ubuntu_packages(packages, pkgs)
+              # merge packages debian style
```

Comment:
a) Do we have "debian_style" packages in all circumstances? Or should we do a if check here?
b) should we name the method "assemble_deb_style_packages" and "assemble_ubuntu_style_packages"?

Reply:
a) yes, currently it is always assembled, but only used for ESM repos.
I agree this is not the best thing to do performance wise.
Although, we only speak about rearranging the data in memory and not re-downloading it.

b) sounds good. We might even want to go further and think about a more descriptive name for the data-structure that is generated.